File: backend/NourishNet/Models/DonationHistory.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace NourishNet.Models
{
    public class DonationHistory
    {
        [Key]
        public int DonationId { get; set; }
        public string RecipientId { get; set; }

        [ForeignKey("RecipientId")]
        public Recipient Recipient { get; set; }

        public int FoodListingId { get; set; }

        [ForeignKey("FoodListingId")]
        public FoodListing FoodListing { get; set; }

        public string DataReceived { get; set; }
        

    }
}

File: frontend/NourishNet/Models/DonationHistory.ts
import { Recipient } from "./Recipient"
import { FoodListing } from "./FoodListing"

export interface DonationHistory {
    donationId: number ,
    recipientId: string,
    recipient: Recipient,
    foodListingId: number | null,
    foodListing: FoodListing | null
    dataRecieved: string        //methna dateRecieved kiyla hadenna one , backend eke mata wardila data kiyla daala hinda methanth daanna una 
}